Enable modulemap on iPhone

adetaylor opened the pull request:
#### 0f6b7ca13c78a2e83e04713ca5532a801cf8beae
<pre>
Enable modulemap on iPhone
<a href="https://bugs.webkit.org/show_bug.cgi?id=296118">https://bugs.webkit.org/show_bug.cgi?id=296118</a>
<a href="https://rdar.apple.com/156040652">rdar://156040652</a>

Reviewed by Richard Robinson.

This enables the production of module.modulemap files for the latest SDK on iOS
for WTF and bmalloc. These files describe how to interpret bmalloc and WTF as
clang modules, and that can influence the way downstream targets get built. We
are therefore enabling it platform-by-platform to give us time to react to
any unforeseen surprises.

* Configurations/WebKitTargetConditionals.xcconfig:
* Source/WTF/Configurations/WTF.xcconfig:
* Source/bmalloc/Configurations/bmalloc.xcconfig:

Canonical link: <a href="https://commits.webkit.org/297730@main">https://commits.webkit.org/297730@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2c701c8475007343ae403a65ee9028372537b9a3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112531 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32263 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22741 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118729 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62988 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32915 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40826 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85642 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/36270 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115478 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26266 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101240 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65944 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25564 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19372 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62489 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/105045 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95657 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19445 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121951 "Built successfully") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/111145 "Found 1 new JSC stress test failure: stress/catch-from-function-inlining-simd-function.js.ftl-eager-no-cjit (failure)") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39605 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29495 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94509 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39987 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97471 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94249 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24088 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39362 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17169 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/35690 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39493 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44981 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/135377 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39130 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36380 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42465 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40871 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7ca351540e9fda504fe4e3c707f702afcf4621f5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48179

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2c701c8475007343ae403a65ee9028372537b9a3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48179

adetaylor committed:
Enable modulemap on iPhone
https://bugs.webkit.org/show_bug.cgi?id=296118
rdar://156040652

Reviewed by Richard Robinson.

This enables the production of module.modulemap files for the latest SDK on iOS
for WTF and bmalloc. These files describe how to interpret bmalloc and WTF as
clang modules, and that can influence the way downstream targets get built. We
are therefore enabling it platform-by-platform to give us time to react to
any unforeseen surprises.

* Configurations/WebKitTargetConditionals.xcconfig:
* Source/WTF/Configurations/WTF.xcconfig:
* Source/bmalloc/Configurations/bmalloc.xcconfig:

Canonical link: https://commits.webkit.org/297730@main
